File: app-kotlin/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/investigation/mapsmenu/mapdisplay/data/models/PoiModel.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.mapsmenu.mapdisplay.data.models;

import android.graphics.PointF;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class PoiModel {

    private int id = -1;
    private String name = "";
    private PoiType type = PoiType.values()[0];
    @Nullable
    private PointF point = null;

    public PoiModel() {
        setId(-1);
        setName("");
    }

    public PoiModel(@Nullable PoiModel tempPoiData) {
        if(tempPoiData == null) { return; }

        setId(tempPoiData.getId());
        setName(tempPoiData.getName());
        setType(tempPoiData.getType());
        setPoint(tempPoiData.getPoint());
    }

    public void setPoint(PointF tempPoiData) {
        this.point = tempPoiData;
    }

    public PointF getPoint() {
        return point;
    }

    public PoiModel(int id, String name, int type, float x, float y) {
        setId(id);
        setName(name);
        setType(type);
        setPoint(new PointF(x, y));
    }


    public void setName(String name) {
        this.name = name;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setType(PoiType type) {
        this.type = type;
    }

    public void setType(int type) {
        setType(PoiType.values()[type]);
    }

    public PoiType getType() {
        return type;
    }

    public boolean hasName() {
        return !this.name.equals("");
    }

    public boolean hasId() {
        return this.id >= -1;
    }

    @NonNull
    public String toString() {
        return "\n\t\t[Room ID: " + id + "] [Room Name: " + name + "] [Room Name: " + type + "] [Room points: " + point + "]";
    }

    public boolean isReady() {
        return point != null;
    }

    public void print() {
        Log.d("Maps", id + " " + name + " " + type + " " + point);
    }
}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/investigation/mapsmenu/mapdisplay/data/models/PoiModel.kt
package com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.mapsmenu.mapdisplay.data.models

import android.graphics.PointF
import android.util.Log

class PoiModel {
    var id: Int = -1
    var name: String = ""
    @JvmField
    var type: PoiType = PoiType.entries[0]
    @JvmField
    var point: PointF? = null

    constructor() {
        id = -1
        name = ""
    }

    constructor(tempPoiData: PoiModel?) {
        if (tempPoiData == null) {
            return
        }

        id = tempPoiData.id
        name = tempPoiData.name
        type = tempPoiData.type
        point = tempPoiData.point
    }

    constructor(id: Int, name: String, type: Int, x: Float, y: Float) {
        this.id = id
        this.name = name
        setType(type)
        point = PointF(x, y)
    }


    fun setType(type: Int) {
        this.type = PoiType.entries[type]
    }

    fun hasName(): Boolean {
        return !name.isEmpty()
    }

    fun hasId(): Boolean {
        return this.id >= -1
    }

    override fun toString(): String {
        return "\n\t\t[Room ID: $id] [Room Name: $name] [Room Name: $type] [Room points: $point]"
    }

    val isReady: Boolean
        get() = point != null

    fun print() {
        Log.d("Maps", "$id $name $type $point")
    }
}
